interface/src/app/node/node.js

```diff
@@ -92,5 +92,204 @@ angular.module('adage.node', [
+                },
+                'size': {
+                  'field': 'highlight',
+                  'type': 'nominal',
+                  'scale': {'range': [10, 30]},
+                  'legend': false
+                }
+              }
+            };
+            embedSpec.spec = vlSpec;
+          }; // end of setEmbedSpec() definition.
+
+          // Enhance experiment objects and put them into $scope.experiments.
+          response.data.objects.forEach(function(element) {
+            enhanceExpData(element);
+            $scope.experiments.push(element);
+          });
+          // Sort experiments by range in descending order:
+          $scope.experiments.sort(function(e1, e2) {
+            return e2.range - e1.range;
+          });
+
+          // Add vega-lite widget for each experiment:
+          var embedSpec = {'mode': 'vega-lite', 'actions': false};
+          var sample2index = {};
+          // Function that toggles "highlight" flag of an experiment's samples:
+          var toggleSampleHighlights = function(expIndex) {
+            var currExp = $scope.experiments[expIndex];
+            var j, sid, arrIndex, val;
+            for (j = 0; j < currExp.sample_set.length; ++j) {
+              sid = $scope.experiments[i].sample_set[j];
+              arrIndex = sample2index[sid];
+              val = embedSpec.spec.data.values[arrIndex].highlight;
+              embedSpec.spec.data.values[arrIndex].highlight = !val;
+            }
+          };
+          setEmbedSpec(embedSpec, sample2index); // Initialize embedSpec.
+          // For each experiment, toggle the highlight flags (to true) for
+          // all of its samples, embed the vega-lite widget, and toggle these
+          // highlight flags again (to false). This procedure ensures that
+          // only activitity values in current experiment will be highlighted
+          // in the widget.
+          for (var i = 0, n = $scope.experiments.length; i < n; ++i) {
+            toggleSampleHighlights(i);
+            vg.embed('#exp-' + i, embedSpec, function(error, result) { });
+            toggleSampleHighlights(i);
+          }
+          // Clear $scope.queryStatus to indicate that the view is ready!
+          $scope.queryStatus = '';
+        }, // end of success function.
+        function error(err) {
+          $log.error('Failed to get activities: ' + err.statusText);
+          $scope.queryStatus = 'Failed to get related experiments from server';
+        }); // end of $http.get().then() chaining.
+
+      // Event handler when user clicks "+" or "-" icon in front of
+      // an experiment.
+      $scope.toggleExpansion = function(exp) {
+        exp.isExpanded = !exp.isExpanded;
+        // Get samples data if they are not available yet.
+        if (exp.isExpanded && !exp.samples) {
+          exp.sampleStatus = 'Connecting to the server ...';
+          var sampleURI = '/api/v0/sample/?experiment=' + exp.accession;
```

Comment:
So, it looks like ``$http.get()`` can take a second argument after the url, which is a ``config`` object {}. In this object, you can specify these querystring parameters inside a ``params`` key instead of adding them to the end of the url: https://docs.angularjs.org/api/ng/service/$http#usage